OrbitLinuxTracing/PerfEventOpen.cpp

```diff
@@ -1,11 +1,5 @@
 #include "PerfEventOpen.h"
 
-#include <OrbitBase/Logging.h>
```

Comment:
Please include all header files that you actually need below.

Reply:
Is there a good common practice for this? All these header files are already included in PerfEventOpen.h, that's why they were removed. Trying to include all and only the necessary ones in the cpp file is always a failure as the compiler can't help as it already sees the ones included in the header.

Reply:
What does "is always a failure" mean here? All header files that are needed in a .cpp file should be included there and it shouldn't matter whether they are already included in the header file (which should only be true for those that are also needed in the header file).

Reply:
I just meant that in my experience it always start like a good thing to do, but as most headers needed by the cpp file are usually also needed by the header file, the compiler doesn't help you with also adding the theoretically needed headers to the cpp file. This means that trying to achieve this is a big manual effort (every time I want to use a function I need to check whether that header is already included by the cpp file or only by the header file) that doesn't really give anything (since the cpp always includes its header anyway), and as the cpp file grows it is inevitable to start forgetting to include some headers anyway. 
But this was just my reasoning behind this deletion. I added them back (the correct ones for the change).

Reply:
You can also have a lot of additional transitive includes, not just what's listed in the header file of the .cpp file you are editing. That doesn't mean one should just give up and only include new stuff when it doesn't build anymore. There are automatic tools for this we can look into (clang-include-fixer), but in the mean time we should just try to do a good job with keeping includes as close to what's needed in the .cpp file.